web_src/js/features/admin/selfcheck.js

```diff
@@ -0,0 +1,31 @@
+import {toggleElem} from '../../utils/dom.js';
+import {POST} from '../../modules/fetch.js';
+
+const {appSubUrl} = window.config;
+
+export async function initAdminSelfCheck() {
+  const elCheckByFrontend = document.querySelector('.page-content.admin .self-check-by-frontend');
+  if (!elCheckByFrontend) return;
+
+  const elContent = document.querySelector('.page-content.admin .admin-setting-content');
```

Comment:
Called too early, move it where it's used first.

Reply:
Not early, I think it is the proper place. The next blocks are for "fetch" request, then elContent must be used there.

Reply:
If I don't move, does it affect anything? performance? readability? maintainability?

I think current code is more readable and I definitely will keep it as it.

Reply:
I think it's common sense to declare a variable as close as possible to where it's being used as otherwise one could think that there must be a specific reason to assign it before the `fetch`, for example.

Reply:
In my mind, it is a common sense to prepare the necessary page elements ahead before other logic. Then when refactoring, we can clearly know this page uses `'.page-content.admin .admin-setting-content'` , and all other `init` functions also query `.page-content.xxx`  at the beginning.

And one more reason, `elContent` is a container. In the future all problem checks will be done by a fetch and the content will be directly written into the elContent. So I do not see any reason to move it.